cnd/src/storage/db/tables.rs

```diff
@@ -508,14 +496,34 @@ pub struct InsertableBtcDaiOrder {
     pub order_id: i32,
     pub quantity: Text<Satoshis>,
     pub price: Text<Erc20Amount>,
+    pub open: Text<Satoshis>,
+    pub closed: Text<Satoshis>,
+    pub settling: Text<Satoshis>,
+    pub failed: Text<Satoshis>,
+    pub cancelled: Text<Satoshis>,
 }
 
 impl InsertableBtcDaiOrder {
-    pub fn new(order_fk: i32, quantity: asset::Bitcoin, price: Erc20Quantity) -> Self {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        order_fk: i32,
+        quantity: asset::Bitcoin,
+        price: Erc20Quantity,
+        open: asset::Bitcoin,
+        closed: asset::Bitcoin,
+        settling: asset::Bitcoin,
+        failed: asset::Bitcoin,
+        cancelled: asset::Bitcoin,
+    ) -> Self {
         Self {
```

Comment:
📣 a check should be done to ensure that the quantities make sense (the sum of the state quantities equals the `quantity`).

Reply:
Instead I moved the initialisation into the constructor; `open` is set to `quantity`, all other states are initialized with `ZERO`. That way the initial state is ensured.